pkg/executor/wasm/executor_test.go

```diff
@@ -0,0 +1,71 @@
+package wasm
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/bacalhau-project/bacalhau/pkg/executor"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+// Mocking necessary dependencies and structures
+
+type mockHandler struct {
+	mock.Mock
+}
+
+func (m *mockHandler) active() bool {
+	args := m.Called()
+	return args.Bool(0)
+}
+
+func (m *mockHandler) run(ctx context.Context) {
+	_ = m.Called(ctx)
+}
+
+type mockExecutor struct {
+	mock.Mock
+}
+
+func (m *mockExecutor) Get(executionID string) (*mockHandler, bool) {
+	args := m.Called(executionID)
+	return args.Get(0).(*mockHandler), args.Bool(1)
+}
+
+func (m *mockExecutor) Put(executionID string, handler *mockHandler) {
+	_ = m.Called(executionID, handler)
+}
+
+func TestFailingRequestedMemGreaterThan4GB(t *testing.T) {
+	e, err := NewExecutor()
+
+	assert.Nil(t, err)
```

Comment:
Rather than assert, most of our tests use `t.Require().*`. So in this case `t.Require().NoError(err)`

Reply:
okay, quite new to both

Reply:
@rossjones i just checked, the `t.Require()` is not present in testing.T

Reply:
also `assert.Nil` is used in the codebase

Reply:
looks more like `unittest` in python. Not sure if its really necessary for BD test. but sure i ill make the changes

Reply:
Done

Reply:
I get it s.Require does assert.Error not nil and t.FailNow(). Make sense